assets/src/dashboard/app/views/previewStory/index.js

```diff
@@ -0,0 +1,202 @@
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+/**
+ * WordPress dependencies
+ */
+import { __, sprintf } from '@wordpress/i18n';
+
+/**
+ * External dependencies
+ */
+import { useContext, useEffect, useRef, useState } from 'react';
+import PropTypes from 'prop-types';
+import styled from 'styled-components';
+
+/**
+ * Internal dependencies
+ */
+import { Modal, TypographyPresets, Button } from '../../../components';
+import { WPBODY_ID, BUTTON_TYPES } from '../../../constants';
+import dashboardTheme from '../../../theme';
+import { Close as CloseIcon } from '../../../icons';
+import { StoryPropType } from '../../../types';
+import { useResizeEffect } from '../../../utils';
+import { ApiContext } from '../../api/apiProvider';
+
+const CLOSE_BUTTON_SIZE = {
+  HEIGHT: 30,
+  WIDTH: 30,
+};
+const PREVIEW_CONTAINER_ID = 'previewContainer';
+
+const CloseButton = styled.button`
+  align-self: flex-end;
+  color: ${({ theme }) => theme.colors.white};
+  margin-top: 20px;
+  margin-right: 11px;
+  width: ${CLOSE_BUTTON_SIZE.WIDTH}px;
+  height: ${CLOSE_BUTTON_SIZE.HEIGHT}px;
+  border: ${({ theme }) => theme.borders.transparent};
+  background-color: transparent;
+  z-index: 10;
+
+  &:hover {
+    cursor: pointer;
+  }
+`;
+
+const IframeContainer = styled.div`
+  width: ${({ dimensions }) => `${dimensions.width}px`};
+  height: ${({ dimensions }) =>
+    `${dimensions.height - CLOSE_BUTTON_SIZE.HEIGHT}px`};
+`;
+
+const HelperText = styled.p`
+  ${TypographyPresets.Large};
+  margin: 0;
+  padding-bottom: 20px;
+  color: ${({ theme }) => theme.colors.white};
+`;
+
+const HelperContainer = styled.div`
+  bottom: 0;
+  width: 100%;
+  min-height: 90vh;
+  display: flex;
+  align-items: center;
+  justify-content: center;
+  flex-direction: column;
+`;
+
+const PreviewStory = ({ story, handleClose, isTemplate }) => {
+  const {
+    state: {
+      stories: { previewMarkup, isLoading, error },
+    },
+    actions: {
+      storyApi: { createStoryPreviewFromTemplate, clearStoryPreview },
+    },
+  } = useContext(ApiContext);
+
+  const containerRef = useRef(document.getElementById(WPBODY_ID));
+
+  const [modalDimensions, setModalDimensions] = useState({
+    width: containerRef.current?.offsetWidth || window.innerWidth,
+    height: containerRef.current?.offsetHeight || window.innerHeight,
+  });
+  const [previewError, setPreviewError] = useState(error.message?.title);
+
+  useEffect(() => {
+    const iframeContainer = document.getElementById(PREVIEW_CONTAINER_ID);
+    if (previewMarkup.length > 0 && iframeContainer) {
+      let iframe = document.createElement('iframe');
+      iframeContainer.appendChild(iframe);
+      iframe.setAttribute('style', 'height:100%;width:100%;border:none;');
+      iframe.contentWindow.document.open();
+      iframe.contentWindow.document.write(previewMarkup);
+      iframe.contentWindow.document.close();
+    }
+  }, [previewMarkup]);
```

Comment:
I think this might append a new iframe child to the `iframeContainer` every time the effect runs. might need to add a cleanup function returned by the effect.

Reply:
good call! am clearing innerHTML on clean up now.